service/resource/deployment/deployment.go

```diff
@@ -66,10 +66,12 @@ func (r Resource) newMainDeployment(cluster azuretpr.CustomObject) (Deployment,
 		"apiLoadBalancerCidr":           cluster.Spec.Azure.VirtualNetwork.LoadBalancer.APICIDR,
 		"etcdLoadBalancerCidr":          cluster.Spec.Azure.VirtualNetwork.LoadBalancer.EtcdCIDR,
 		"ingressLoadBalancerCidr":       cluster.Spec.Azure.VirtualNetwork.LoadBalancer.IngressCIDR,
+		"mastersCustomConfig":           cluster.Spec.Azure.Masters,
```

Comment:
As you can see `params` is passed to `convertParams` below, which isn't quite ready yet for receiving nested object types. I'll try to update it today.

For the reference current implementation:

```go
// convertParameters converts the map into the structure used by the Azure API.
func convertParameters(inputs map[string]interface{}) map[string]interface{} {
	params := make(map[string]interface{}, len(inputs))
	for key, val := range inputs {
		params[key] = struct {
			Value interface{}
		}{
			Value: val,
		}
	}

	return params
}
```

Reply:
I created issue for that: https://github.com/giantswarm/azure-operator/issues/31

Reply:
Issue was closed as `convertParams` is working with nested objects.